src/videojs-hls.js

```diff
@@ -736,8 +736,14 @@ videojs.Hls.prototype.drainBuffer = function(event) {
   this.segmentParser_.flushTags();
 
   tags = [];
+  segment.preciseTimestamp = null;
+
   while (this.segmentParser_.tagsAvailable()) {
     tags.push(this.segmentParser_.getNextTag());
+    if (!tags[tags.length - 1].counted && tags[tags.length-1].pts) {
```

Comment:
Are you doing this inside the loop in case PTS values are not ordered ascending? I don't think that's necessary.